backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepositoryCache.java

```diff
@@ -21,24 +22,35 @@ public class GitMacheteRepositoryCache implements IGitMacheteRepositoryCache {
 
   private final IGitCoreRepositoryFactory gitCoreRepositoryFactory;
 
-  private static final java.util.Map<Tuple2<Path, Path>, GitMacheteRepository> gitMacheteRepositoryCache = new WeakHashMap<>();
+  private static io.vavr.collection.Map<Tuple2<Path, Path>, SoftReference<GitMacheteRepository>> gitMacheteRepositoryCache = HashMap
```

Comment:
`io.vavr.collection.Map` can be normally imported IIRC according to our rules, only `java.util.Map` needs to be explicitly qualified